privacyidea/lib/eventhandler/base.py

```diff
@@ -356,10 +398,19 @@ def check_condition(self, options):
                 if not ','.join([tok.get_serial() for tok in token_objects]):
                     return False
 
-        if "tokenrealm" in conditions and tokenrealms:
+        if CONDITION.TOKENREALM in conditions and tokenrealms:
             res = False
             for trealm in tokenrealms:
-                if trealm in conditions.get("tokenrealm").split(","):
+                if trealm in conditions.get(CONDITION.TOKENREALM).split(","):
+                    res = True
+                    break
+            if not res:
+                return False
+
+        if CONDITION.TOKENRESOLVER in conditions and tokenresolvers:
```

Comment:
Same question as above: What if the token has no resolvers?

Reply:
s.o.